models/Element/Service.php

```diff
@@ -1148,9 +1148,8 @@ public static function getValidKey($key, $type)
             $key = ltrim($key, '. ');
         }
 
-        $key = mb_substr($key, 0, 255);
-
-        return $key;
+        // key should not end with space after cut
+        return trim(mb_substr($key, 0, 255));
```

Comment:
Should we use `rtrim` if we only care about removing whitespace at the end?

```suggestion
        return rtrim(mb_substr($key, 0, 255));
```

Otherwise, I think we should adapt the comment.

Reply:
Good point, thanks,
my guess on adapt a comment, better be sure that there are no spaces on start and end, 
no matter what is going on before

Reply:
i already see a trim on 
https://github.com/pimcore/pimcore/pull/15347/files?w=1#diff-f1c5d50aa07c3685f76a2947c0c3859e2582f8b683ebadf175679eb4aae78785R1126

is the problem only occuring when a key is exceeding 255 characters and the substr cut exactly after a space?
eg.
`Service::getValidKey('FbeF7zEVFcIidFWEHqLNAqw3eZShMApdX0AYSxDv7ROwm5nFLgdLKU9aJ8ZCcIEVlUYHrbtnAAIbjFRDF6HDIR6S7hcOetexdaJaBSPXScQWVhzkO0nu6Eyu7FTAIpbx03gwKS3WXdlivTOVZ5LimYoYSKfGBh1SKBAEpPpOqWQeboxNLZrtuLUF9ZXhXj0E1ic0qBmslbHEA8C44JTxPbrDFfbOnA7nPt6Q1AQZe26IDVULnKwcV49eOXJG3H YMYQGD', 'object');`

Reply:
Yes, bc of mb_substr() function we rarely get this exception on a big strings.
ex: I'm importing few millions of products and using title for a key and got ~100 random exceptions on that,
currently I'm using hack `trim(Service::getValidKey($id.$title, 'object'))`

Reply:
never mind, it was a false positive